### PR TITLE
Try to cache the theme json file reading

### PR DESCRIPTION
**Pull request by cbravobernal**
[Try to fix #44619](https://github.com/WordPress/gutenberg/issues/44619#issuecomment-1266866044)

Testing instructions:
https://github.com/WordPress/gutenberg/issues/44619#issuecomment-1266866044

Not working yet 😢 

Trac ticket: https://core.trac.wordpress.org/ticket/56736

### CODE REVIEW

File: src/wp-includes/class-wp-theme-json-resolver.php [hunk 1]
```diff
@@ -177,20 +186,23 @@ public static function get_theme_data( $deprecated = array(), $options = array()
...
 			 *
 			 * @param WP_Theme_JSON_Data Class to access and update the underlying data.
 			 */
-			$theme_json      = apply_filters( 'theme_json_theme', new WP_Theme_JSON_Data( $theme_json_data, 'theme' ) );
-			$theme_json_data = $theme_json->get_data();
-			static::$theme   = new WP_Theme_JSON( $theme_json_data );
+
+			static::$theme_json_data = $theme_json_data;
+			$theme_json              = apply_filters( 'theme_json_theme', new WP_Theme_JSON_Data( $theme_json_data, 'theme' ) );
+			$theme_json_data         = $theme_json->get_data();
```
Comment (cbravobernal) [inline src/wp-includes/class-wp-theme-json-resolver.php]: If we set `static::$theme_json_data` after these two lines. Then the theme json file does not apply.

File: src/wp-includes/class-wp-theme-json-resolver.php [hunk 2]
```diff
@@ -177,20 +186,23 @@ public static function get_theme_data( $deprecated = array(), $options = array()
...
+			static::$theme           = new WP_Theme_JSON( $theme_json_data );
```
Comment (cbravobernal) [inline src/wp-includes/class-wp-theme-json-resolver.php]: We should test it with child themes.

**Comment by cbravobernal** [comment]
cc/ @oandregal, would this approach be correct?

**Comment by ockham** [comment]
Rebased.

**Comment by ockham** [comment]
I think this is working now 🎉 

I was about to implement file caching for the parent theme (see https://github.com/WordPress/wordpress-develop/pull/3401#discussion_r986868064), but then it occurred to me that we might solve this slightly differently:

We can add a single `$theme_json_file_cache` static var, and use it from within `static::read_json_file()`.

I'll give that a try.

**Comment by cbravobernal** [comment]
https://github.com/WordPress/wordpress-develop/pull/3408 does the work, so we can close this one.

**Pull request closed without merging**